src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -667,12 +708,17 @@ private void ThrowIfNoFunctionResultsAdded(IList<ChatMessage>? messages)
     /// <param name="iteration">The iteration number of how many roundtrips have been made to the inner client.</param>
     /// <param name="functionCallIndex">The 0-based index of the function being called out of <paramref name="callContents"/>.</param>
     /// <param name="captureExceptions">If true, handles function-invocation exceptions by returning a value with <see cref="FunctionInvocationStatus.Exception"/>. Otherwise, rethrows.</param>
+    /// <param name="isStreaming">Whether the function calls are being processed in a streaming context.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests.</param>
     /// <returns>A value indicating how the caller should proceed.</returns>
-    private async Task<FunctionInvocationResult> ProcessFunctionCallAsync(
-        List<ChatMessage> messages, ChatOptions options, List<FunctionCallContent> callContents,
-        int iteration, int functionCallIndex, bool captureExceptions, CancellationToken cancellationToken)
+    protected virtual async Task<FunctionInvocationResult> ProcessFunctionCallAsync(
```

Comment:
I thought we were reverting this to be private again?

Reply:
This is a must need for our specialization as we construct the `context` and make the `AutoInvocationFilter` work.

https://github.com/microsoft/semantic-kernel/pull/11678/files#diff-9ef864b76a36e09adf87b47254aa4be3902332f986f08289ce94ea930970febaR291

Reply:
moved to `private`